File: backend/nest/src/modules/viagem/entities/viagem.entity.ts
import { Prop, raw, Schema, SchemaFactory } from '@nestjs/mongoose';
import mongoose from 'mongoose';

@Schema()
export class Viagem {

  @Prop({ type: mongoose.Schema.Types.ObjectId, required: false, auto: true })
  _id?: mongoose.Types.ObjectId;

  @Prop({ type: String, required: false })
  custo: string;

  @Prop({ type: [mongoose.Schema.Types.ObjectId], ref: 'usuario', required: false})
  id_usuarios?: string[];

  @Prop({ type: String, required: false })
  origem: string;

  @Prop({ type: String, required: false })
  destino: string;

  @Prop({ type: String, required: false })
  data_hora_partida: Date;

  @Prop({ type: String, required: false })
  data_hora_chegada: Date;

  @Prop({ type: Number, required: false })
  quantidade_de_vagas: number;

  @Prop({ type: String, required: false })
  nome_prestador: string;

  @Prop({ type: Boolean, required: false })
  finalizada?: boolean;
}

export const ViagemSchema = SchemaFactory.createForClass(Viagem);

File: frontend/mobile/lib/features/viagem/domain/entities/viagem_entity.dart
// lib/features/viagem/domain/entities/viagem.dart
class Viagem {
  final String id;
  final String custo;
  final String origem;
  final String destino;
  final String dataHoraPartida;
  final String dataHoraChegada;
  final int quantidadeDeVagas;
  final String nomePrestador;
  final List<String> idUsuarios;

  Viagem({
    required this.id,
    required this.custo,
    required this.origem,
    required this.destino,
    required this.dataHoraPartida,
    required this.dataHoraChegada,
    required this.quantidadeDeVagas,
    required this.nomePrestador,
    required this.idUsuarios,
  });

  factory Viagem.fromJson(Map<String, dynamic> json) {
    return Viagem(
      id: json['_id'],
      custo: json['custo'],
      origem: json['origem'],
      destino: json['destino'],
      dataHoraPartida: json['data_hora_partida'],
      dataHoraChegada: json['data_hora_chegada'],
      quantidadeDeVagas: json['quantidade_de_vagas'],
      nomePrestador: json['nome_prestador'],
      idUsuarios: List<String>.from(json['id_usuarios']),
    );
  }
}
